pulp_ansible/app/migrations/0035_deprecation_content.py

```diff
@@ -0,0 +1,80 @@
+# Generated by Django 2.2.24 on 2021-07-07 17:46
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+def migrate_data_from_old_model_to_new_model_up(apps, schema_editor):
+    """ Move objects from AnsibleCollectionDeprecated to NewAnsibleCollectionDeprecated."""
+    AnsibleCollectionDeprecated = apps.get_model('ansible', 'AnsibleCollectionDeprecated')
+    NewAnsibleCollectionDeprecated = apps.get_model('ansible', 'NewAnsibleCollectionDeprecated')
+    for deprecated_collection in AnsibleCollectionDeprecated.objects.all():
+        NewAnsibleCollectionDeprecated(
+            namespace=deprecated_collection.collection.namespace,
+            name=deprecated_collection.collection.name,
+            pulp_type="ansible.collection_deprecation",
+            collection_id=deprecated_collection.collection_id,
+            repository_id=deprecated_collection.repository_version.repository.pk,
+            version_added_id=deprecated_collection.repository_version.pk,
+            version_added_number=deprecated_collection.repository_version.number,
+        ).save()
+
+
+def migrate_data_from_old_model_to_new_model_down(apps, schema_editor):
+    """ Move objects from NewAnsibleCollectionDeprecated to AnsibleCollectionDeprecated."""
+    AnsibleCollectionDeprecated = apps.get_model('ansible', 'AnsibleCollectionDeprecated')
+    NewAnsibleCollectionDeprecated = apps.get_model('ansible', 'NewAnsibleCollectionDeprecated')
+    RepositoryVersionContentDetails = apps.get_model('core', 'RepositoryVersionContentDetails')
+    for deprecation in NewAnsibleCollectionDeprecated.objects.all():
+        AnsibleCollectionDeprecated(
+            collection_id=deprecation.collection_id,
+            repository_version_id=deprecation.version_added_id,
+        ).save()
+        deprecation.version_memberships.all().delete()
+    RepositoryVersionContentDetails.objects.filter(content_type=deprecation.pulp_type).delete()
+
+def delete_deprecation_type(apps, schema_editor):
+    """ Delete deprecation content type."""
+    Content = apps.get_model('core', 'Content')
+    Content.objects.filter(pulp_type="ansible.collection_deprecation").delete()
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('core', '0068_add_timestamp_of_interest'),
+        ('ansible', '0034_handle_jsonfield_warnings'),
+    ]
+
+    operations = [
+        migrations.RunPython(
+            code=migrations.RunPython.noop,
+            reverse_code=delete_deprecation_type,
```

Comment:
I don't fully understand what's happening in this step, but shouldn't all delete actions happen after `migrate_data_from_old_model_to_new_model_up`?